core/src/test/scala/org/apache/pekko/persistence/r2dbc/journal/RuntimePluginConfigSpec.scala

```diff
@@ -0,0 +1,322 @@
+import pekko.persistence.r2dbc.query.scaladsl.R2dbcReadJournal
+import pekko.persistence.r2dbc.state.scaladsl.R2dbcDurableStateStore
+import pekko.persistence.state.scaladsl.GetObjectResult
+import pekko.persistence.typed.PersistenceId
+import pekko.persistence.typed.scaladsl.Effect
+import pekko.persistence.typed.scaladsl.EventSourcedBehavior
+import pekko.persistence.typed.scaladsl.RetentionCriteria
+import pekko.stream.scaladsl.Sink
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.Inside
+import org.scalatest.freespec.AnyFreeSpecLike
+import org.slf4j.LoggerFactory
+
+object RuntimePluginConfigSpec {
+
+  trait EventSourced {
+    import EventSourced._
+
+    def configKey: String
+    def database: String
+
+    lazy val config: Config =
+      ConfigFactory
+        .load(
+          ConfigFactory
+            .parseString(
+              s"""
+              $configKey = $${pekko.persistence.r2dbc}
+              $configKey = {
+                connection-factory {
+                  database = "$database"
+                }
+
+                journal.$configKey.connection-factory = $${$configKey.connection-factory}
+                journal.use-connection-factory = "$configKey.connection-factory"
+                query.$configKey.connection-factory = $${$configKey.connection-factory}
+                query.use-connection-factory = "$configKey.connection-factory"
+                snapshot.$configKey.connection-factory = $${$configKey.connection-factory}
+                snapshot.use-connection-factory = "$configKey.connection-factory"
+              }
+              """
+            )
+            .withFallback(TestConfig.unresolvedConfig)
+        )
+
+    def apply(persistenceId: String): Behavior[Command] =
+      EventSourcedBehavior[Command, String, String](
+        PersistenceId.ofUniqueId(persistenceId),
+        "",
+        (state, cmd) =>
+          cmd match {
+            case Save(text, replyTo) =>
+              Effect.persist(text).thenRun(_ => replyTo ! Done)
+            case ShowMeWhatYouGot(replyTo) =>
+              replyTo ! state
+              Effect.none
+            case Stop =>
+              Effect.stop()
+          },
+        (state, evt) => Seq(state, evt).filter(_.nonEmpty).mkString("|"))
+        .withRetention(RetentionCriteria.snapshotEvery(1, Int.MaxValue))
+        .withJournalPluginId(s"$configKey.journal")
+        .withJournalPluginConfig(Some(config))
+        .withSnapshotPluginId(s"$configKey.snapshot")
+        .withSnapshotPluginConfig(Some(config))
+  }
+  object EventSourced {
+    sealed trait Command
+    case class Save(text: String, replyTo: ActorRef[Done]) extends Command
+    case class ShowMeWhatYouGot(replyTo: ActorRef[String]) extends Command
+    case object Stop extends Command
+  }
+
+  trait DurableState {
+    def typedSystem: ActorSystem[_]
+    def configKey: String
+    def database: String
+
+    lazy val config: Config =
+      ConfigFactory
```

Comment:
again I'm confused by need for load and parseString

Reply:
Same comment as https://github.com/apache/pekko-persistence-r2dbc/pull/194#discussion_r1994276716.